### PR TITLE
Milestone/deposit

### PR DESCRIPTION
**Pull request by ECWireless**
### By Feb 27th

- [x]  Background
- [x]  Navigation Bar
- [x]  Router
- [x]  Add Sidebar

### By March 2nd

- [x]  Main Card Component
- [x]  Deposit Field
- [x]  Stats

### By March 10th

- [x] Display wPOKT balance
- [x] Add input and submit functionality
- [x] Approve functionality
- [x] Stake functionality
- [x] Add `notify` notifications

Note: this is without custom styles for the `onboardjs` modal and `notify` notifications

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -10,18 +10,39 @@ import Navigation from 'components/Navigation';
 import Sidebar from 'components/Sidebar';
 import Wrapper from 'components/Wrapper';
 
+// Context
```
Comment (Evalir) [inline src/App.tsx]: Let's remove these comments :)

File: src/App.tsx [hunk 2]
```diff
@@ -10,18 +10,39 @@ import Navigation from 'components/Navigation';
...
+import { Web3Context } from 'contexts/Web3';
+
 // Views
```
Comment (Evalir) [inline src/App.tsx]: Ditto

File: src/App.tsx [hunk 3]
```diff
@@ -10,18 +10,39 @@ import Navigation from 'components/Navigation';
...
 import Deposit from 'views/Deposit';
 
 const App: React.FC = () => {
+  // Context
```
Comment (Evalir) [inline src/App.tsx]: Ditto

File: src/App.tsx [hunk 4]
```diff
@@ -10,18 +10,39 @@ import Navigation from 'components/Navigation';
...
+  const {
+    onboard,
+    provider,
+    // wallet,
+    // notify
```
Comment (Evalir) [inline src/App.tsx]: Ditto

File: src/App.tsx [hunk 5]
```diff
@@ -10,18 +10,39 @@ import Navigation from 'components/Navigation';
...
+  } = React.useContext(Web3Context);
+
   // State
```
Comment (Evalir) [inline src/App.tsx]: Ditto

File: src/App.tsx [hunk 6]
```diff
@@ -10,18 +10,39 @@ import Navigation from 'components/Navigation';
...
+
+  async function readyToTransact() {
+    if (!provider) {
+      const walletSelected = await onboard?.walletSelect();
+      if (!walletSelected) return false;
+    }
+
+    const ready = await onboard?.walletCheck();
+    return ready;
+  }
```
Comment (Evalir) [inline src/App.tsx]: Hmm, does this check if a user is connected to a wallet? I'm getting the feeling this should be an effect.

File: src/components/Navigation/index.tsx [hunk 7]
```diff
@@ -61,8 +67,8 @@ const Navigation: React.FC<INavigation> = ({ setSidebar }) => {
         </ul>
       </StyledNavigationItems>
       <Flex align={'center'}>
-        <StyledConnectWalletButton>
-          Connect
+        <StyledConnectWalletButton onClick={readyToTransact}>
+          {address ? `${address?.slice(0, 5)}...` : 'Connect'}
```
Comment (Evalir) [inline src/components/Navigation/index.tsx]: This is nice :) But we can probably use [this](https://github.com/aragon/client/blob/master/src/web3-utils.js#L256-L282) utility!

File: src/contexts/Balance/Provider.tsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import { BalanceContext } from './Context';
+
+// Utils
+import { getERC20Contract } from 'utils';
+
+export const BalanceProvider: React.FC = ({ children }) => {
+  const { address, provider } = React.useContext(Web3Context);
+  const [wpoktBalance, setWpoktBalance] = React.useState<string>();
+
+  const getBalance = async (provider: Provider, tokenAddress: string, userAddress: string): Promise<string> => {
```
Comment (Evalir) [inline src/contexts/Balance/Provider.tsx]: We should probably wrap this in an `useCallback` as well, or move it out to not re-declare it 😄

File: src/contexts/Balance/Provider.tsx [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    try {
+      const balance = await tokenContract.balanceOf(userAddress);
+      return balance.toString();
+    } catch (e) {
+      console.error(e);
+      return '0';
+    }
+  };
+
+  const fetchBalances = React.useCallback(
```
Comment (Evalir) [inline src/contexts/Balance/Provider.tsx]: also, just to double check: are we calling every `React`  import with the dot syntax instead of importing it? (e.g `React.useEffect`) It's fine if we're doing so, I like it—But let's double-check make sure it's like that on the whole codebase 💯

File: src/contexts/Balance/Provider.tsx [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  );
+
+  React.useEffect((): (() => void) => {
+    if (address && provider) {
+      fetchBalances(address, provider);
+      const refreshInterval = setInterval(() => fetchBalances(address, provider), 10000);
+      return () => clearInterval(refreshInterval);
+    }
+    return () => null;
+    // eslint-disable-next-line
```
Comment (Evalir) [inline src/contexts/Balance/Provider.tsx]: Uhm haha should this be here?

File: src/contexts/Balance/index.ts [hunk 11]
```diff
@@ -0,0 +1,2 @@
+export { BalanceContext } from './Context';
+export { BalanceProvider } from './Provider';
```
Comment (Evalir) [inline src/contexts/Balance/index.ts]: It's honestly kinda weird to me that this is split on 3 files... Feels a bit bloaty—having declarations and exports on one file is not bad.

File: src/contexts/Balance/types.ts [hunk 12]
```diff
@@ -0,0 +1,3 @@
+export interface ContextValues {
+  wpoktBalance?: any;
+}
```
Comment (Evalir) [inline src/contexts/Balance/types.ts]: Actually four haha... Could we merge all of this into one file?

File: src/contexts/Web3/Provider.tsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+        if (wallet.provider) {
+          setWallet(wallet);
+
+          const ethersProvider = new ethers.providers.Web3Provider(wallet.provider);
+
+          provider = ethersProvider;
+          signer = getSigner(ethersProvider);
+
+          if (wallet.name) {
+            window.localStorage.setItem('selectedWallet', wallet.name);
```
Comment (Evalir) [inline src/contexts/Web3/Provider.tsx]: We should probably prefix the key with something like `WPOKT_` and make it a constant. What do you think?

File: src/libs/connector.ts [hunk 14]
```diff
@@ -0,0 +1,79 @@
+import Notify, { API } from 'bnc-notify';
+import Onboard from 'bnc-onboard';
+import getRpcUrl from 'libs/rpc';
+import { Subscriptions, API as OnboardAPI } from 'libs/types';
+
+const networkId = 4;
+const rpcUrl = getRpcUrl(networkId);
+const dappId = process.env.REACT_APP_ONBOARD_API_KEY;
```
Comment (Evalir) [inline src/libs/connector.ts]: These are all constants right? If so, I'd write them in uppercase and snake case:
```suggestion
const NETWORK_ID = 4;
const RPC_URL = getRpcUrl(networkId);
const DAPP_ID = process.env.REACT_APP_ONBOARD_API_KEY;
```

File: src/libs/rpc.ts [hunk 15]
```diff
@@ -0,0 +1,9 @@
+export default function getRpcUrl(networkId: number): string | undefined {
+  switch (networkId) {
+    case 1:
+      return process.env.MAINNET_RPC;
```
Comment (Evalir) [inline src/libs/rpc.ts]: We should have a `.env.example` with all these variables!

File: src/libs/signer.ts [hunk 16]
```diff
@@ -0,0 +1,36 @@
+import { ethers } from 'ethers';
+import Web3 from 'web3';
+
+export function getAddress(): Promise<string> {
+  // await window.ethereum.enable();
+  const provider = new ethers.providers.Web3Provider(Web3.givenProvider);
+  const signer = provider.getSigner();
+  return signer.getAddress();
+}
```
Comment (Evalir) [inline src/libs/signer.ts]: Should this function create a provider each time? This feels like it should be fetched from the wallet context

File: src/utils/bignumber.ts [hunk 17]
```diff
@@ -0,0 +1,5 @@
+import BigNumber from 'bignumber.js';
+
+BigNumber.config({ DECIMAL_PLACES: 100, EXPONENTIAL_AT: 1e9 });
```
Comment (Evalir) [inline src/utils/bignumber.ts]: We don't really need [`BigNumber libraries anymore`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt)—Unless we really need decimal operations.

File: src/utils/index.ts [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+import ERC20ABI from 'abis/ERC20.json';
+import TokenGeyserABI from 'abis/TokenGeyser.json';
+
+export const bnToDec = (bn: BigNumber, decimals = 18): number => {
+  return bn.dividedBy(new BigNumber(10).pow(decimals)).toNumber();
+};
+
+export const decToBn = (dec: number, decimals = 18): BigNumber => {
+  return new BigNumber(dec).multipliedBy(new BigNumber(10).pow(decimals));
+};
```
Comment (Evalir) [inline src/utils/index.ts]: I'm a bit nervous of converting `BN`s to numbers... I'd rather keep them as strings, as ethereum numbers are big enough to make Javascript numbers [overflow](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_VALUE).

File: src/utils/index.ts [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+      // or you could use onclick for when someone clicks on the notification itself
+      onclick: () => window.open(`https://rinkeby.etherscan.io/tx/${transaction.hash}`),
+    };
+  });
+
+  emitter.on('txSent', console.log);
+  emitter.on('txConfirmed', console.log);
+  emitter.on('txSpeedUp', console.log);
+  emitter.on('txCancel', console.log);
+  emitter.on('txFailed', console.log);
```
Comment (Evalir) [inline src/utils/index.ts]: I don't know if this should be enabled for production. Can we hide this behind a `development` flag?

File: src/views/Deposit/components/EnterAmount/components/index.ts [hunk 20]
```diff
@@ -72,13 +72,25 @@ export const StyledDepositInputContainer = styled.div`
...
+
+    &:disabled {
+      cursor: not-allowed !important;
+
+      &:hover {
+        transform: translateY(-50%) !important;
+      }
+
+      &:active {
+        transform: translateY(-50%) !important;
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/components/index.ts]: Oof, why are we using `!important` here? We should definitely try _not_ to do this haha

File: src/views/Deposit/components/EnterAmount/index.tsx [hunk 21]
```diff
@@ -17,7 +22,89 @@ import {
...
+    if (address && provider && !isDisabled && farmSelected) {
+      const allowance = await getAllowance(address, TOKEN_GEYSER_ADDRESS, WPOKT_ADDRESS, provider);
+      if (+decToBn(+wpoktInputValue) > +allowance) {
+        setIsApproved(true);
+      } else {
+        setIsApproved(false);
+      }
+    }
+  };
+  useApproval();
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/index.tsx]: Yup, we can make it a custom hook and just return a function wrapped in a `useCallback`.

File: src/views/Deposit/components/EnterAmount/index.tsx [hunk 22]
```diff
@@ -17,7 +22,89 @@ import {
...
+
+  // Effects
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/index.tsx]: :)

File: src/views/Deposit/components/EnterAmount/index.tsx [hunk 23]
```diff
@@ -17,7 +22,89 @@ import {
...
+  React.useEffect(() => {
+    if (wpoktInputValue === '' || wpoktInputValue === '0' || !farmSelected) {
+      setIsDisabled(true);
+    } else {
+      setIsDisabled(false);
+    }
+  }, [wpoktInputValue, farmSelected]);
+
+  // Handlers
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/index.tsx]: :)

File: src/views/Deposit/components/EnterAmount/index.tsx [hunk 24]
```diff
@@ -17,7 +22,89 @@ import {
...
+          } else {
+            if (notify) {
+              getNotification(notify, response);
+              setWpoktInputValue('');
+              setFarmSelected(false);
+            }
+          }
+        }
+      }
+    }
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/index.tsx]: In terms of logic, this is a bit hard to follow due to all the nesting... We could probably try to flatten it a bit by inverting the logic somewhat, e.g if any of the values checked on the first if is missing, we execute the first case, else we can just check by case.

File: src/views/Deposit/components/EnterAmount/index.tsx [hunk 25]
```diff
@@ -17,7 +22,89 @@ import {
...
+  };
+
+  const onMaxValue = () => {
+    setWpoktInputValue(bnToDec(new BigNumber(wpoktBalance)).toString());
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/index.tsx]: This can probably overflow due to the decimal conversion. 100% sure.

File: src/views/Deposit/components/EnterAmount/index.tsx [hunk 26]
```diff
@@ -29,8 +116,12 @@ export const EnterAmount: React.FC = () => {
         <StyledHeaderRight>
           <StyledLine />
           <div id={'wallet-balance'}>
-            <P2 color={colors.white}>Wallet balance: 2345.926562 wPOKT</P2>
-            <StyledMaxButton>
+            <P2 color={colors.white}>
+              {wpoktBalance
+                ? `Wallet balance: ${bnToDec(new BigNumber(wpoktBalance))} wPOKT`
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/index.tsx]: Same—can have overflow issues

File: src/views/Deposit/components/InfoCard/components/SmallInfoCard.tsx [hunk 27]
```diff
@@ -23,7 +23,33 @@ export const SmallInfoCard: React.FC<ISmallInfoCard> = ({ statContent, statFill,
...
+              background: transparent;
+              border: none;
+
+              &:hover {
+                cursor: pointer;
+              }
+            `}
+          >
+            <MultiplierSvg />
+          </button>
```
Comment (Evalir) [inline src/views/Deposit/components/InfoCard/components/SmallInfoCard.tsx]: This button `css` is repeated, so we should probably make it a component :)

**Review by Evalir — COMMENTED** [review]
Bit of a quick review, but looking good! Some comments:

File: src/App.tsx (hunk 1 above)
Comment (ECWireless) [reply src/App.tsx]: Fixed in 4bf8642d6e8b1d3eab7e08c97ee4512e7d289920

File: src/App.tsx (hunk 2 above)
Comment (ECWireless) [reply src/App.tsx]: Fixed in 4bf8642d6e8b1d3eab7e08c97ee4512e7d289920

File: src/App.tsx (hunk 3 above)
Comment (ECWireless) [reply src/App.tsx]: Fixed in 4bf8642d6e8b1d3eab7e08c97ee4512e7d289920

File: src/App.tsx (hunk 4 above)
Comment (ECWireless) [reply src/App.tsx]: Fixed in 4bf8642d6e8b1d3eab7e08c97ee4512e7d289920

File: src/App.tsx (hunk 5 above)
Comment (ECWireless) [reply src/App.tsx]: Fixed in 4bf8642d6e8b1d3eab7e08c97ee4512e7d289920

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 22 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Fixed in 4bf8642d6e8b1d3eab7e08c97ee4512e7d289920

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 23 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Fixed in 4bf8642d6e8b1d3eab7e08c97ee4512e7d289920

File: src/views/Deposit/components/EnterAmount/components/index.ts (hunk 20 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/components/index.ts]: Yeah, `!important` wasn't even necessary here to begin with 😅. Fixed in 8ecf9d4304d5a3f2a6e62a8485b8eb3f1af69f06

File: src/views/Deposit/components/InfoCard/components/SmallInfoCard.tsx (hunk 27 above)
Comment (ECWireless) [reply src/views/Deposit/components/InfoCard/components/SmallInfoCard.tsx]: Fixed in c1035d70b1e2f4659a79cba790c85b905ed8d5f2

File: src/contexts/Balance/Provider.tsx (hunk 9 above)
Comment (ECWireless) [reply src/contexts/Balance/Provider.tsx]: Yup, just double-checked, and it's consistent throughout!

File: src/contexts/Balance/Provider.tsx (hunk 10 above)
Comment (ECWireless) [reply src/contexts/Balance/Provider.tsx]: Whoops, yeah, shouldn't be there. Fixed in fcdfa813a94ff817747368a3058c107b850ab8c5

File: src/contexts/Balance/index.ts (hunk 11 above)
Comment (ECWireless) [reply src/contexts/Balance/index.ts]: Fixed in a0f4757d01ce1b4ff2b283fd8e81c6bacc09054f

File: src/contexts/Balance/types.ts (hunk 12 above)
Comment (ECWireless) [reply src/contexts/Balance/types.ts]: Fixed in a0f4757d01ce1b4ff2b283fd8e81c6bacc09054f

File: src/components/Navigation/index.tsx (hunk 7 above)
Comment (ECWireless) [reply src/components/Navigation/index.tsx]: Fixed in c36207998e0878960b97d5673b755d1ff8cd2869

File: src/libs/rpc.ts (hunk 15 above)
Comment (ECWireless) [reply src/libs/rpc.ts]: Fixed in 4e7b000ac48a48779622f01b7a40c52c35a50f45

File: src/libs/connector.ts (hunk 14 above)
Comment (ECWireless) [reply src/libs/connector.ts]: Fixed in 4e7b000ac48a48779622f01b7a40c52c35a50f45

File: src/contexts/Web3/Provider.tsx (hunk 13 above)
Comment (ECWireless) [reply src/contexts/Web3/Provider.tsx]: Fixed in 4e7b000ac48a48779622f01b7a40c52c35a50f45

File: src/utils/index.ts (hunk 18 above)
Comment (ECWireless) [reply src/utils/index.ts]: I guess what would be a better way of displaying `bigint` as a decimal on the frontend? One thought is doing something like:

```
export const bigintToDec = (bigint: bigint): number => {
  const convertedNumber = Number(bigint / BigInt(1e9));
  return convertedNumber / 1e9;
};
```

File: src/App.tsx (hunk 6 above)
Comment (ECWireless) [reply src/App.tsx]: Apparently, it's meant to be called simply before each transaction (to make sure nothing has changed since the last transaction): https://docs.blocknative.com/onboard#walletcheck

Although, now that I think about it, maybe the whole `readyToTransact` function should just be moved into `utils`, then called before each transaction?

Comment (ECWireless) [reply src/App.tsx]: Actually, I think I see what you're saying... Lemme make a quick commit

Comment (ECWireless) [reply src/App.tsx]: Fixed in 910b900a7fa76c963e13cff322c250ca782c7b16

File: src/contexts/Balance/Provider.tsx (hunk 8 above)
Comment (ECWireless) [reply src/contexts/Balance/Provider.tsx]: Fixed in 5d379a4f41e2ee7a24388f1d0303aacf5e27d59b

File: src/utils/index.ts (hunk 18 above)
Comment (Evalir) [reply src/utils/index.ts]: To display them, we can just convert them to String and use [token-amount](https://github.com/aragon/token-amount/) to format them correctly. :)

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 21 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Fixed in 856de0c7c7a7109633b56c9cc98309dbdf193b48

File: src/libs/signer.ts (hunk 16 above)
Comment (ECWireless) [reply src/libs/signer.ts]: Yeah, fair enough. I think this may just be a web3 boilerplate helper function. But I think I can just remove it since, `onboardjs` provides an address in the wallet context.

Comment (ECWireless) [reply src/libs/signer.ts]: Fixed in a59ac9b57bade73a60925b772d6b4323f3221b3d

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 24 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Fixed in 856de0c7c7a7109633b56c9cc98309dbdf193b48

File: src/utils/index.ts (hunk 19 above)
Comment (ECWireless) [reply src/utils/index.ts]: I'm actually realizing that `nofity` isn't really necessary for this project, since I can just use `ethers.wait()` to launch the pending transaction modal. Will be adding those modals after this PR is merged tho. Fixed in dcb25cb4373e610cbf87ea9ccf00727c0819c0ad

File: src/utils/bignumber.ts (hunk 17 above)
Comment (ECWireless) [reply src/utils/bignumber.ts]: Fixed in 6cd632d7c520d20ea5f02a83b1ea1b4da83ad696

File: src/utils/index.ts (hunk 18 above)
Comment (ECWireless) [reply src/utils/index.ts]: Fixed in 6cd632d7c520d20ea5f02a83b1ea1b4da83ad696

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 25 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Fixed in 6cd632d7c520d20ea5f02a83b1ea1b4da83ad696

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 26 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Fixed in 6cd632d7c520d20ea5f02a83b1ea1b4da83ad696

File: src/components/Navigation/components/index.ts [hunk 28]
```diff
@@ -49,6 +56,22 @@ export const StyledConnectWalletButton = styled.button`
     height: ${13 * GU}px;
     width: ${56 * GU}px;
   `}
+
+  ${(props) =>
+    props.connected &&
+    css`
```
Comment (Evalir) [inline src/components/Navigation/components/index.ts:56]: Do we need this `css` function?

File: src/views/Deposit/components/EnterAmount/index.tsx [hunk 29]
```diff
@@ -22,87 +21,65 @@ import {
...
-        } else {
-          if (notify) {
-            getNotification(notify, response);
-          }
-        }
+    readyToTransact(onboard, provider);
+    if (address && signer && !isDisabled && farmSelected) {
+      if (isApproved) {
+        const dummyToken = new TokenAmount('1', 0);
+        const convertedAmount = dummyToken.convert(wpoktInputValue, 18).value;
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/index.tsx]: I don't quite get these changes with removing `bignumber.js`—how are we handling user input now for decimals? Are we just converting it to a string and displaying that on the "front", while keeping the bignumber on the "back"? `TokenAmount` should only be used for formatting purposes or converting from rates, so I'm a bit confused. 😅

File: src/views/Deposit/components/EnterAmount/index.tsx [hunk 30]
```diff
@@ -118,7 +95,7 @@ export const EnterAmount: React.FC<IEnterAmount> = ({ farmSelected, setFarmSelec
           <div id={'wallet-balance'}>
             <P2 color={colors.white}>
               {wpoktBalance
-                ? `Wallet balance: ${bnToDec(new BigNumber(wpoktBalance))} wPOKT`
+                ? `Wallet balance: ${new TokenAmount(wpoktBalance, 18, { symbol: 'wPOKT' })}`
```
Comment (Evalir) [inline src/views/Deposit/components/EnterAmount/index.tsx]: Instead of declaring a new object (which I'm surprised works haha), we can also use the static method `TokenAmount.format()` which has basically the same API.

**Review by Evalir — APPROVED** [review]
This overall is looking pretty great, just wanna nail down the math/user input for depositing:

File: src/components/Navigation/components/index.ts (hunk 28 above)
Comment (ECWireless) [reply src/components/Navigation/components/index.ts:56]: This just switches font size between when the button says "Connect" vs "0x12345...abcd". Though maybe it's cleaner to just make the address' `charsLength` 3. Fixed in 6925a12af2c57de963edc809022536cc114a9924

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 30 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Fixed in 19d679215f7ab2d86f679d5811c7201ac9489b2a

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 29 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: I actually was more or less copying these tests to handle decimals: https://github.com/aragon/token-amount/blob/master/src/TokenAmount.test.ts#L26

Bit of a cheat maybe, but it seemed that you could use the convert method to change decimal inputs to "non-formatted" values.

Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Honestly, wasn't sure of a good way to convert decimal inputs to "non-formatted" values if we aren't using `bignumber.js`. But this seems to weirdly do the trick. Do you think this will still cause precision issues, or does it just look confusing haha?

File: src/components/Navigation/components/index.ts (hunk 28 above)
Comment (Evalir) [reply src/components/Navigation/components/index.ts:56]: Oh no, it was fine before! Just was wondering as we were already in a CSS context haha... And now with this new change I think it was better like it was before 😅

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 29 above)
Comment (Evalir) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Yeah, this is definitely confusing haha... I can see this works for converting from decimals to amounts, but if so, let's write a bunch of tests for this to make sure this won't either:
1 - Give wrong values for big or small amounts (a lot of decimals, or a huge amount being given to the input)
2 - be inaccurate at any point, no matter the amount

If we wanna switch to a battle-tested implementation, I'd take a look at [this hook](https://github.com/aragon/upgrade/blob/master/src/components/Migrate/Converter/useInputValidation.ts)—we can probably strip out a bunch of stuff from this and use it, even though it would mean using Ethers's `BigNumber` type (which is fine because we already have it).

Comment (Evalir) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Sorry if I seem very bikesheddy on this part of the app... But really, after we get this form right, everything else is pretty much downhill :)

Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Ah, no worries, no worries. So I guess the crux of this hook is really just using [`EthersUtils.parseUnits()`](https://github.com/aragon/upgrade/blob/67fdd07f411f210daf1fb9e23450bcac8461cd52/src/utils/math-utils.ts#L15), right? If so, I'll just pop that in in place of what I have!

File: src/components/Navigation/components/index.ts (hunk 28 above)
Comment (ECWireless) [reply src/components/Navigation/components/index.ts:56]: Consider it reverted: 87cd257bb6a5f94667d45ff5ffc17d46a59f4772

File: src/views/Deposit/components/EnterAmount/index.tsx (hunk 29 above)
Comment (ECWireless) [reply src/views/Deposit/components/EnterAmount/index.tsx]: Alright, decided to go the battle-tested route: 8a6f067dfaa8a6ae95bce9a65c198df33172dcb4
